pkg/assets/assets.go

```diff
@@ -102,13 +104,30 @@ func assetFromTemplate(name string, tb []byte, data interface{}) (*Asset, error)
 	return &Asset{Name: name, Data: bs}, nil
 }
 
-type FileInfoPredicate func(os.FileInfo) bool
+type FileInfoPredicate func(string, os.FileInfo) bool
 
 // OnlyYaml is a predicate for LoadFilesRecursively filters out non-yaml files.
-func OnlyYaml(info os.FileInfo) bool {
+func OnlyYaml(_ string, info os.FileInfo) bool {
 	return strings.HasSuffix(info.Name(), ".yaml") || strings.HasSuffix(info.Name(), ".yml")
 }
 
+// InstallerFeatureSet returns a predicate for LoadFilesRecursively that filters manifests
+// based on the specified FeatureSet.
+func InstallerFeatureSet(featureSet string) FileInfoPredicate {
+	targetFeatureSet := "Default"
+	if len(featureSet) > 0 {
+		targetFeatureSet = featureSet
+	}
+	return func(path string, info os.FileInfo) bool {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			panic(err)
```

Comment:
ugly, but for our uses, if @tkashem will let you have it, so be it.

Reply:
It's your call, Luis :)

Reply:
Changed to propagate error.